src/python/pants/option/custom_types.py

```diff
@@ -308,9 +308,14 @@ def __repr__(self):
     return '{} {}'.format(self.action, self.val)
 
 
-class GlobExpansionConjunction(enum('conjunction', ['any_match', 'all_match'])):
+class GlobExpansionConjunction(enum(['any_match', 'all_match'], field_name='conjunction')):
```

Comment:
Why do we allow customising the field name at all? Can we just always use `value`? Or if we're worried about namespace clashes, always use `._underlying_string` or something? It would also allow us to get rid of the `underlying` method which has some kind of sketchy "Please don't use this, but if you need to use it" documentation :)

Reply:
I would love to remove the field name. Now I know it is ok to do. I had made a change which did that and reverted it for unknown reasons.

Reply:
Also, the field name can't start with an underscore, you get a `ValueError` in `namedtuple`. I was quite sad.